source/Halibut.Tests/UsageFixture.cs

```diff
@@ -159,5 +159,45 @@ public async Task StreamsCanBeSentWithProgressReporting(ClientAndServiceTestCase
                 progressReported.Should().ContainInOrder(Enumerable.Range(1, 100));
             }
         }
+
+        [Test]
+        [LatestAndPreviousClientAndServiceVersionsTestCases]
+        public async Task OctopusCanSendAndReceiveComplexObjectsWithMultipleDataStreams(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase
+                       .CreateTestCaseBuilder()
+                       .WithStandardServices()
+                       .Build(CancellationToken))
+            {
+                var service = clientAndService.CreateClient<IComplexObjectService>();
+                var requestId = "TestRequestIdentifier";
+                var payload = "ThisIsMyTestString";
+                var childPayload1 = "ChildPayload1";
+                var childPayload2 = "ChildPayload2";
+                
+                var result = service.Process(new ComplexRequest
+                {
+                    RequestId = requestId,
+                    NumberOfCopies = 3,
+                    Payload = DataStream.FromString(payload),
+                    Child = new ComplexObjectChild
+                    {
+                        First = DataStream.FromString(childPayload1),
+                        Second = DataStream.FromString(childPayload2)
+                    }
+                });
+
+                result.RequestId.Should().Be(requestId);
+
+                var payloads = result.Payloads.ToArray();
+                payloads.Length.Should().Be(3);
+                payloads[0].ReadAsString().Should().Be(ComplexObjectService.ProcessPayload(1, payload).ReadAsString());
```

Comment:
Consider making the assert against a string literal, that way the test is easier to read and we don't find out later that shenanigans are at play e.g. `ProcessPayload` has some bug of always returning empty which happens to make the test meaningless.